InterGalacticCurrencyConvertor/src/main/java/materials/MaterialFactory.java

```diff
@@ -0,0 +1,55 @@
+package main.java.materials;
+
+import java.util.HashMap;
+import java.util.Objects;
+
+/**
+ * Keeps track of different types of materials that can be sold InterGalactically.
+ */
+public final class MaterialFactory {
+  private HashMap<String, Material> mMtlMap  = new HashMap<>();
+  private static MaterialFactory mInstance = new MaterialFactory();
+  public static MaterialFactory getInstance() {
+    return mInstance;
+  }
+  public void add(String name, int numUnits, double totalCost) {
+    Objects.requireNonNull(name, "Invalid Material Name");
+    mMtlMap.put(name, new Material(name, numUnits, totalCost));
```

Comment:
You don't covert name to lowercase when you put it in the map as a key. But you use lowercase in  the isAvailable(..) method. Can this cause issues?

Reply:
added it in add, yes was not causing issues because i was doing it in the calling methods of the processor. Removed them in those places and confined them in this class only